rice-middleware/ksb/client-impl/src/main/java/org/kuali/rice/ksb/security/soap/KualiPasswordEncryptor.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2006-2015 The Kuali Foundation
+ *
+ * Licensed under the Educational Community License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.opensource.org/licenses/ecl2.php
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kuali.rice.ksb.security.soap;
+
+import org.apache.wss4j.common.crypto.PasswordEncryptor;
+
+public class KualiPasswordEncryptor implements PasswordEncryptor {
+    @Override
+    public String encrypt(String password) {
+        return password;
+    }
+
+    @Override
+    public String decrypt(String encryptedPassword) {
+        return encryptedPassword;
+    }
+}

```

Comment:
More information: http://coheigea.blogspot.com/2014/02/apache-wss4j-200-part-iv.html

So basically we're getting the keystore information (alias, file, password) form our rice-config.xml.  That happens in CXFWSS4JInInterceptor.getMerlinProperties.  Now our password  we have in that config file is plain text, so the KualiPasswordEncryptor (should really be named PlainTextPasswordEncryptor) is working fine.  

If I understand it correctly, for added security we could use the JasyptPasswordEncryptor  default.  Existing config files should work fine since only passwords of format ENC(encoded encrypted password) are decrypted/encrypted.  In other words, the KualiPassowrdEncryptor only gets exercised when <param name="keystore.password">ENC(r1c3pw)</param>.

BTW, this page mentions that the PBEWithMD5AndTripleDES encryption of JasyptPasswordEncryptor isn't totally secure anymore either. The best way is to implement the PasswordEncryptor pug-able/configurable (via the new configuration tag ConfigurationConstants.PASSWORD_ENCRYPTOR_INSTANCE?) so implementing institutions could swap out the PasswordEncryptor.  But I wonder if implementing institutions would go that far considering the pain password is currently stored on their server which should already have limited access to admins.

My suggestion still is to use the JasyptPasswordEncryptor if possible.